llama_stack/providers/adapters/inference/bedrock/bedrock.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import Tuple, Dict, List, Optional, Union, AsyncGenerator
+
+import boto3
+from botocore.client import BaseClient
+from llama_models.datatypes import SamplingParams
+from llama_models.llama3.api import StopReason, Message, ToolCall, CompletionMessage, ToolParamDefinition, \
+    ToolDefinition, ToolChoice, InterleavedTextMedia, ToolPromptFormat, BuiltinTool
+
+from llama_models.llama3.api.chat_format import ChatFormat
+from llama_models.llama3.api.tokenizer import Tokenizer
+from llama_models.sku_list import resolve_model
+
+from llama_stack.apis.inference import Inference, ChatCompletionResponse, \
+    ChatCompletionResponseEvent, ChatCompletionResponseStreamChunk, ChatCompletionResponseEventType, \
+    ToolCallParseStatus, ToolCallDelta, LogProbConfig, CompletionResponse, CompletionResponseStreamChunk
+
+# mapping of Model SKUs to ollama models
+BEDROCK_SUPPORTED_MODELS = {
+    "Meta-Llama3.1-8B-Instruct": "meta.llama3-1-8b-instruct-v1:0",
+    "Meta-Llama3.1-70B-Instruct": "meta.llama3-1-70b-instruct-v1:0",
+    "Meta-Llama3.1-405B-Instruct": "meta.llama3-1-405b-instruct-v1:0",
+}
+
+
+class BedrockInferenceAdapter(Inference):
+    def __init__(self, url: str) -> None:
+        self.url = url
+
+        self._client = boto3.client("bedrock-runtime")

```

Comment:
how are the auth / api keys retrieved?

Reply:
Right now it was getting it from the environment. Env vars, EC2/ECS credential provider. I'm updating to make it more explicit with the default being getting it from the environment.